modules/ui/src/app/pages/settings/general-settings.component.ts

```diff
@@ -16,52 +16,43 @@
 import {
   Component,
   EventEmitter,
-  Input,
   OnDestroy,
   OnInit,
   Output,
 } from '@angular/core';
 import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
 import { Subject, takeUntil, tap } from 'rxjs';
-import { SystemInterfaces } from '../../services/test-run.service';
 import { OnlyDifferentValuesValidator } from './only-different-values.validator';
 import { CalloutType } from '../../model/callout-type';
 import { CdkTrapFocus, LiveAnnouncer } from '@angular/cdk/a11y';
 import { EventType } from '../../model/event-type';
-import { SettingOption, SystemConfig } from '../../model/setting';
-import { Store } from '@ngrx/store';
-import { AppState } from '../../store/state';
-import { selectSystemConfig } from '../../store/selectors';
-import { createSystemConfig } from '../../store/actions';
-import { AppEffects } from '../../store/effects';
+import { FormKey, SystemConfig } from '../../model/setting';
+import { SettingsStore } from './settings.store';
+import { LoaderService } from '../../services/loader.service';
 
 @Component({
   selector: 'app-general-settings',
   templateUrl: './general-settings.component.html',
   styleUrls: ['./general-settings.component.scss'],
   hostDirectives: [CdkTrapFocus],
+  providers: [SettingsStore],
 })
 export class GeneralSettingsComponent implements OnInit, OnDestroy {
-  @Input() interfaces: SystemInterfaces = {};
-  @Input() hasConnectionSettings = false;
   @Output() closeSettingEvent = new EventEmitter<void>();
-  @Output() reloadInterfacesEvent = new EventEmitter<void>();
   public readonly CalloutType = CalloutType;
   public readonly EventType = EventType;
+  public readonly FormKey = FormKey;
   public settingForm!: FormGroup;
-  public isSubmitting = false;
-  public defaultInternetOption = {
-    key: '',
-    value: 'Not specified',
-  };
+  vm$ = this.settingsStore.vm$;
```

Comment:
Could you please explain what is mean vm? Probably it is better to create name that will self explain.

Reply:
Done